src/Core/Cache/StandardCacheWarmer.php

```diff
@@ -0,0 +1,297 @@
+						'Default',
+						basename($templateFile, '.' . $format)
+					),
+					$renderingContext);
+				$result->add($state, $templateFile);
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Warm up _partialRootPaths_ of the provided RenderingContext's
+	 * TemplatePaths instance. Simple, recursive processing of all
+	 * supported format template files in path(s), compiling only
+	 * the topmost (override) template file if the same template
+	 * exists in multiple partial root paths.
+	 *
+	 * Like other methods, returns a FluidCacheWarmupResult instance
+	 * which can be merged with other result instances.
+	 *
+	 * @param RenderingContextInterface $renderingContext
+	 * @param array $formats
+	 * @return FluidCacheWarmupResult
+	 */
+	protected function warmupPartialRootPaths(RenderingContextInterface $renderingContext, array $formats) {
+		$result = new FluidCacheWarmupResult();
+		$paths = $renderingContext->getTemplatePaths();
+		foreach ($formats as $format) {
+			foreach ($paths->resolveAvailablePartialFiles($format) as $partialFile) {
+				$paths->setFormat($format);
+				$state = $this->warmSingleFile(
+					$partialFile,
+					$paths->getPartialIdentifier(basename($partialFile, '.' . $format)),
+					$renderingContext
+				);
+				$result->add($state, $partialFile);
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Warm up _layoutRootPaths_ of the provided RenderingContext's
+	 * TemplatePaths instance. Simple, recursive processing of all
+	 * supported format template files in path(s), compiling only
+	 * the topmost (override) template file if the same template
+	 * exists in multiple layout root paths.
+	 *
+	 * Like other methods, returns a FluidCacheWarmupResult instance
+	 * which can be merged with other result instances.
+	 *
+	 * @param RenderingContextInterface $renderingContext
+	 * @param array $formats
+	 * @return FluidCacheWarmupResult
+	 */
+	protected function warmupLayoutRootPaths(RenderingContextInterface $renderingContext, array $formats) {
+		$result = new FluidCacheWarmupResult();
+		$paths = $renderingContext->getTemplatePaths();
+		foreach ($formats as $format) {
+			foreach ($paths->resolveAvailableLayoutFiles($format) as $layoutFile) {
+				$paths->setFormat($format);
+				$state = $this->warmSingleFile(
+					$layoutFile,
+					$paths->getLayoutIdentifier(basename($layoutFile, '.' . $layoutFile)),
+					$renderingContext
+				);
+				$result->add($state, $layoutFile);
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Detect all available controller names in provided TemplateRootPaths
+	 * array, returning the "basename" components of controller-template
+	 * directories encountered, as an array.
+	 *
+	 * @param string $templateRootPaths
+	 * @return \Generator
+	 */
+	protected function detectControllerNamesInTemplateRootPaths(array $templateRootPaths) {
+		foreach ($templateRootPaths as $templateRootPath) {
+			foreach ((array) glob(rtrim($templateRootPath, '/') . '/*') as $pathName) {
```

Comment:
`DIRECTORY_SEPARATOR` substituted - I'm not sure about `glob` on Windows but since this is both wrapped in a CLI command, intended for deployment process, and possible to override in custom warmers, I tend to not really care too much about Windows here ;)